pkg/scheduler/scheduler.go

```diff
@@ -281,10 +286,33 @@ func New(client clientset.Interface,
 	sched.StopEverything = stopEverything
 	sched.client = client
 
-	addAllEventHandlers(sched, informerFactory)
+	// Build dynamic client and dynamic informer factory
+	var dynInformerFactory dynamicinformer.DynamicSharedInformerFactory
+	// options.kubeConfig can be nil in tests.
+	if options.kubeConfig != nil {
+		dynClient := dynamic.NewForConfigOrDie(options.kubeConfig)
+		dynInformerFactory = dynamicinformer.NewFilteredDynamicSharedInformerFactory(dynClient, 0, v1.NamespaceAll, nil)
+	}
+
+	addAllEventHandlers(sched, informerFactory, dynInformerFactory, unionedGVKsExcludingNodePod(clusterEventMap))
 	return sched, nil
 }
 
+func unionedGVKsExcludingNodePod(m map[framework.ClusterEvent]sets.String) map[framework.GVK]framework.ActionType {
```

Comment:
I would not exclude anything in this function, instead get the union of all and add explicit case statements that does nothing for the kinds that we always want to handle.

Reply:
Done.